kitsune/wiki/models.py

```diff
@@ -263,39 +279,53 @@ def save(self, *args, **kwargs):
         # which would cause a rollback, which would negate any category changes
         # we make here, so don't worry:
         self._clean_category()
+        self._clean_template_status()
 
         slug_changed = hasattr(self, 'old_slug')
         title_changed = hasattr(self, 'old_title')
 
-        # If the slug changed, we clear out the share link so it gets regenerated.
-        self.share_link = ''
+        templateness_changed = (
+            title_changed and
+            # If the title was a template before, and is not now, or vice-versa
+            (self.title.startswith(TEMPLATE_TITLE_PREFIX) !=
+             self.old_title.startswith(TEMPLATE_TITLE_PREFIX)))
```

Comment:
I didn't look at it enough to know why being case insensitive helped.

I think I would add some code to the old_title case to cover the template
thing. That seems least invasive.
On Jul 23, 2015 3:51 PM, "Mike Cooper" <notifications@github.com> wrote:

> In kitsune/wiki/models.py
> <https://github.com/mozilla/kitsune/pull/2547#discussion_r35363555>:
>
> >
> >          slug_changed = hasattr(self, 'old_slug')
> >          title_changed = hasattr(self, 'old_title')
> >
> > -        # If the slug changed, we clear out the share link so it gets regenerated.
> > -        self.share_link = ''
> > +        templateness_changed = (
> > +            title_changed and
> > +            # If the title was a template before, and is not now, or vice-versa
> > +            (self.title.startswith(TEMPLATE_TITLE_PREFIX) !=
> > +             self.old_title.startswith(TEMPLATE_TITLE_PREFIX)))
>
> Nice catch. I'm not terrible sure what to do here. My two thoughts are to
> either
>
>    1. make the old_slug and old_title checks no longer case insensitive.
>    2. Add special handling to the part that sets old_title to watch for
>    case changes in the template prefix.
>
> Do you have opinions on either of these? I'm not sure why it was case
> insensitive to being with, so I'm not sure if changing that would be a good
> idea. It would be simpler though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/kitsune/pull/2547/files#r35363555>.
>